### ISSUE TITLE
Using arcface with other networks?

### ISSUE THREAD

**Issue by s-rog** [4uiiurz1/keras-arcface#8]
Hello,

I'm trying to use this with an existing network that ends with | GAP -> dropout -> softmax | and I'm not sure where to put the arc/cosface layer. I've already tried many combinations (replacing the softmax) but couldn't get any to work. Accuracy is always 0 and loss fluctuates but doesn't go down.

Any help is appreciated!

**Comment by 4uiiurz1** [comment]
@s-rog 
Could you paste your code?

**Comment by s-rog** [comment]
I think I figured it out, I'm not supposed to connect a GAP layer to it. Instead I have to use a flattened local feature map with bn like your code in archs.py, correct?

**Comment by 4uiiurz1** [comment]
@s-rog
Maybe the same problem as https://github.com/4uiiurz1/keras-arcface/issues/7 .

**Comment by s-rog** [comment]
```
block7a_project_conv (Conv2D)   (None, 4, 4, 320)    368640      block7a_se_excite[0][0]          
block7a_project_bn (BatchNormal (None, 4, 4, 320)    1280        block7a_project_conv[0][0]       
top_conv (Conv2D)               (None, 4, 4, 1280)   409600      block7a_project_bn[0][0]         
top_bn (BatchNormalization)     (None, 4, 4, 1280)   5120        top_conv[0][0]                   
top_activation (Activation)     (None, 4, 4, 1280)   0           top_bn[0][0]
```                     

These are the last few layers of my pre-trained network (efficientnet). I need to add:

1. batch normalization
2. dropout
3. flatten
4. dense
5. batch normalization
6. arcface

Is that correct?

**Comment by 4uiiurz1** [comment]
@s-rog 
Your architecture is maybe right.
Please refer to https://github.com/4uiiurz1/keras-arcface/issues/7 !
